core/domain/classifier_services.py

```diff
@@ -127,6 +129,117 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+def handle_trainable_states(exploration, state_names):
+    """Creates ClassifierTrainingJobModel instances for all the state names
+    passed into the function. If this function is called with version number 1,
+    we are creating jobs for all trainable states in the exploration. Otherwise,
+    a new job is being created for the states where retraining is required.
+
+    Args:
+        exploration: Exploration. The Exploration domain object.
+        state_names: list(str). List of state names.
+    """
+    job_dicts_list = []
```

Comment:
Actually I think the dict here is important, in order to distinguish between python dict representations of jobs, vs job domain objects.